scrapy/commands/crawl.py

```diff
@@ -54,5 +62,14 @@ def run(self, args, opts):
             raise UsageError("running 'scrapy crawl' with more than one spider is no longer supported")
         spname = args[0]
 
-        self.crawler_process.crawl(spname, **opts.spargs)
+        d = self.crawler_process.crawl(spname, **opts.spargs)
+
+        def _crawl_error(_exc):
+            global exception
```

Comment:
Is it a way to have a mutable variable in a closure (i.e. something what `nonlocal` keyword does in Python 3)? If so, I think it is better to use another workaround, e.g. using a list of length 1 and changing its first item.